File: TwoSum2.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int i = 0;
        int j = numbers.length - 1;
        int co = target - numbers[i];
        while (i < j) {
            if (co == numbers[j]) {
                int ans[] = {i+1,j+1};
                return ans;
            }
            if (co < numbers[j]) {
                j--;
            } else {
                i++;
                co = target - numbers[i];
            }
        }
        return new int[2];
    }
}

File: Array/TwoSum2.scala
object Solution {
    def twoSum(numbers: Array[Int], target: Int): Array[Int] = {
        var i = 0;
        var j = numbers.length -1;
        
        while (i < j) {
            if (numbers(i) + numbers(j) == target) {
                return Array(i+1,j+1);
            } else if (numbers(i) + numbers(j) > target) {
                j-=1;
            } else {
                i+=1;
            }
        }
        return new Array(2);
    }
}
